src/controllers/userController.js

```diff
@@ -0,0 +1,113 @@
+const User = require('../models/User');
+const { StatusCodes } = require('http-status-codes');
+const paginateAndSort = require('../utils/paginationSorting');
+const { NotFoundError } = require('../errors');
+const authenticationMiddleware = require('../middleware/authentication');
+const UnauthenticatedError = require('../errors/unauthenticated');
+
+// Get all users with pagination
+const getUsers = [
+  authenticationMiddleware,
```

Comment:
Oksana, thank you for hard working! You are doing a great job. I have a small concern about using middleware it implementing here (you are also using [chaining](public/uploads/Screenshot 2024-04-16 023316.png) in the userRouter.js). I am not sure that it is a best practice. Maybe we need chaining in the app.js, because  `authenticationMiddleware` will be using in different paths. @benochi what is your advice?

Reply:
Yep. this should be moved to the Routes file.  Unauth error might also be best served as a resposne in the middleware itself.

Reply:
Yep, if this has been taken care of please let me know.  Files changed appears to still ahve it in the controller and it should be in the routes.